app.py

```diff
@@ -45,6 +47,30 @@ def init_app(self):
                                      self.create_panel, 
                                      {"type": "panel", 
                                       "short_name": "shotgun_panel"})
+
+    @property
+    def emitter(self):
+        """
+        The Emitter object utilized by the app. Any QSignals emitted
+        by the app will come from this object.
+        """
+        return self._emitter
+
+    @property
+    def context_change_allowed(self):
+        """
+        Specifies that context changes are allowed.
+        """
+        return True
+
+    def post_context_change(self, old_context, new_context):
+        """
+        Runs after a successful change of context.
+
+        :param old_context: The context prior to the context change.
+        :param new_context: The new context after the context change.
+        """
+        self.emitter.context_changed.emit(old_context, new_context)
```

Comment:
I've altered the behavior to only navigate the panel on context change if it isn't visible. This mimics the previous behavior even when a panel widget is being reused rather than rebuilt (like in current tk-maya). Once we decide on panel behavior during context change, we can use what I built here and tinker with the logic as necessary.